src/test/java/hw6_jdi/JdiTest.java

```diff
@@ -0,0 +1,94 @@
+package hw6_jdi;
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.jdi.light.elements.init.PageFactory;
+import hw6_jdi.entities.MetalsColors;
+import hw6_jdi.entities.MetalsColorsJ;
+import hw6_jdi.entities.User;
+import hw6_jdi.forms.MetalsColorsForm;
+import hw6_jdi.jdiPages.LeftMenuSection;
+import org.testng.Assert;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static hw6_jdi.JdiSite.*;
+
+public class JdiTest {
+
+    @BeforeSuite(alwaysRun = true)
+    public void setUp() {
+        PageFactory.initElements(JdiSite.class);
+    }
+
+    @Test(dataProviderClass = JdiDataProvider.class, dataProvider = "provideDataSetForMetalColortest")
+    public void MetalsAndColorsPageTest(MetalsColorsJ dataSet) {
+        open();
+        jdiHomePage.login(User.ROMAN);
+        jdiHomePage.userName.is().text(User.ROMAN.getFullName());  //Assert.assertEquals(getUserName(), User.ROMAN.getFullName());
+        jdiHomePage.headerSection.clickMetalsColorsMenu();
+        jdiMetalsColorsPage.checkOpened();                          //Assert.assertEquals(getTitle(), "Metal and Colors");
+
+        jdiMetalsColorsPage.metalsColorsForm.fillForm(new MetalsColors(dataSet.getRadio1(), dataSet.getRadio2(),
+                dataSet.getElementsAsString(), dataSet.getColor(), dataSet.getMetals(), dataSet.getVegetablesAsString()));
+
+        //Assertion block for selected elements in Metals & Colors Form
+        Assert.assertEquals(jdiMetalsColorsPage.metalsColorsForm.radio_odd.getValue(), dataSet.getRadio1());
```

Comment:
You have lots of assertions here, your test looks quite heavy because of them. You can create the method in the PO layer, for example.

Reply:
I removed these "assertion block" to separate method in test class